api/internal/managers/app/config/config.go

```diff
@@ -245,8 +246,8 @@ func getConfigValueFromChildItem(item kotsv1beta1.ConfigItem, childItem kotsv1be
 	return configValue
 }
 
-// filterAppConfig filters out disabled groups and items based on their 'when' condition
-func filterAppConfig(config kotsv1beta1.Config) (kotsv1beta1.Config, error) {
+// filterAppConfig filters out disabled groups, items based on their 'when' condition and hidden items
+func filterAppConfig(config kotsv1beta1.Config, filterHiddenItems bool) (kotsv1beta1.Config, error) {
```

Comment:
We can't just filter the hidden items altogether because the validation method needs the ability to validate hidden required items:
- https://github.com/replicatedhq/embedded-cluster/blob/f925f036f6f32685b008972e4c37f079fcca5962/api/internal/managers/app/config/config_test.go#L2815-L2844